src/github.com/matrix-org/dendrite/clientapi/routing/login.go

```diff
@@ -82,12 +183,10 @@ func Login(
 
 		util.GetLogger(req.Context()).WithField("user", r.User).Info("Processing login request")
 
-		// r.User can either be a user ID or just the localpart... or other things maybe.
-		localpart := r.User
```

Comment:
This ensures that `localpart` ("username" in "@username:matrix.org") is actually a localpart and not the full userID ("@username:matrix.org"). By just passing `r` to your two functions and then pulling `r.User`, you're not guaranteeing that `r.User` is just a localpart. 

I would keep this `if` statement to pull out the localpart if necessary and then pass that directly to your functions (`handlePasswordLogin`, `handleTokenLogin`).